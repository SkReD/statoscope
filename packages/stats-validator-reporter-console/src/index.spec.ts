import os from 'os';
import Validator from '../../../packages/stats-validator';
import Reporter from './';

const output: string[][] = [];
let consoleLogSpy: jest.SpyInstance<void, string[]> | null = null;

beforeEach(() => {
  consoleLogSpy = jest
    .spyOn(global.console, 'log')
    .mockImplementation((...args: string[]) => {
      output.push(
        args.map((o) =>
          String(o).replace(process.cwd(), '<pwd>').replace(os.tmpdir(), '<tmp>'),
        ),
      );
    });
});

afterEach(() => {
  consoleLogSpy?.mockRestore();
  output.length = 0;
});

test('should work', async () => {
  const validator = new Validator({
    plugins: [[require.resolve('../../stats-validator-plugin-webpack'), 'webpack']],
    rules: {
      'webpack/restricted-modules': ['error', [/\/src\//]],
      'webpack/restricted-packages': ['error', ['foo']],
    },
  });
  const result = await validator.validate(
    require.resolve('../../../test/bundles/v5/simple/stats-prod.json'),
  );
  const reporter = new Reporter();
  await reporter.run(result);

  expect(output).toMatchSnapshot();
});
