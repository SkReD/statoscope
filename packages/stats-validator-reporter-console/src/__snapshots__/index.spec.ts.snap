// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should work 1`] = `
[
  [
    "input.json",
  ],
  [
    "a5ca8b0ac9b2c1315153",
  ],
  [
    "error  Module ./simple/src/index.ts + 3 modules should not be used  webpack/restricted-modules",
  ],
  [
    "error  Module ./simple/src/index2.ts + 1 modules should not be used  webpack/restricted-modules",
  ],
  [
    "error  Module ./simple/src/sub-module.ts should not be used  webpack/restricted-modules",
  ],
  [
    "error  Module ./simple/src/index2.css should not be used  webpack/restricted-modules",
  ],
  [
    "error  Module ./simple/src/test.txt should not be used  webpack/restricted-modules",
  ],
  [
    "error  Module ./simple/src/dyn.ts should not be used  webpack/restricted-modules",
  ],
  [
    "error  Module ./simple/src/statoscope.png should not be used  webpack/restricted-modules",
  ],
  [
    "error  Module ./simple/src/index.ts should not be used  webpack/restricted-modules",
  ],
  [
    "error  Module ./simple/src/index2.ts should not be used  webpack/restricted-modules",
  ],
  [
    "error  foo@1.0.0 should not be used  webpack/restricted-packages",
  ],
  [
    "Instances:",
  ],
  [
    "- simple/node_modules/foo  1.0.0",
  ],
  [
    "",
  ],
  [
    "10 problems (10 errors, 0 warnings)",
  ],
]
`;
